source/h2_connection.c

```diff
@@ -433,120 +435,172 @@ static void s_outgoing_frames_task(struct aws_channel_task *task, void *arg, enu
         "Outgoing frames task acquired message with %zu bytes available",
         msg->message_data.capacity - msg->message_data.len);
 
-    /* Track number of frames encoded, just used for logging */
-    size_t num_frames_encoded = 0;
+    /* Write as many frames from outgoing_frames_queue as possible. */
+    if (s_encode_outgoing_frames_queue(connection, &msg->message_data)) {
+        goto error;
+    }
+
+    /* If outgoing_frames_queue emptied, then write as many DATA frames from outgoing_streams_list as possible. */
+    if (aws_linked_list_empty(outgoing_frames_queue)) {
+        if (s_encode_data_from_outgoing_streams(connection, &msg->message_data)) {
+            goto error;
+        }
+    }
+
+    if (msg->message_data.len) {
```

Comment:
it's not really "new code" from here to the end of the function. I moved big chunks of code out into their own functions (`s_encode_outgoing_frames_queue()` and `s_encode_data_from_outgoing_streams()`) and it's messing up the diffs